src/Enable.Extensions.Queuing.AzureServiceBus/Internal/AzureServiceBusQueueClient.cs

```diff
@@ -123,19 +130,36 @@ protected override void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    _messageReceiver.CloseAsync()
-                        .GetAwaiter()
-                        .GetResult();
-
-                    _messageSender.CloseAsync()
-                        .GetAwaiter()
-                        .GetResult();
+                    // Use Monitor to prevent race conditions
+                    Monitor.Enter(_queues);
```

Comment:
This will block a thread until a lock is acquired. Would it be better to use `TryEnter` (as it currently stands, the code is no different than using `lock` directly)? Or is there a better construct that we can use?

Reply:
I've gone with lock as it is simpler and should certainly work.